clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1497,4 +1505,16 @@ public void handleResponse(AbstractResponse response) {
             }
         }
     }
+
+    private boolean isGroupMetadataMisMatch(Errors error) {
+        return error == Errors.FENCED_INSTANCE_ID
```

Comment:
User will have to handle this exception properly by either:
1. aborting the transaction as recommended
2. fail the entire application to be more strict
3. other cases as they see fit

So no matter how to handle it, the error shall not be ignored. If we are throwing 3 different types of exceptions, user would as well need to catch 3 different types of exceptions IMHO

Reply:
My expectation for illegal generation and unknown member id is that it can be more or less ignored. The user should abort the transaction, but then continue after rejoining the group. The instance fenced error means a new instance of the application has been started somewhere and the application should be stopped.

By the way, this is why I suggested writing some example code which shows what we consider to be _proper_ handling. This will give us a better idea if the handling is reasonable, awkward, or incomplete.

Reply:
That's true, we also had a jira to track it here: https://issues.apache.org/jira/browse/KAFKA-9447